api/urls_v2.py

```diff
@@ -50,6 +50,12 @@ class AplusRouter(ExtendedDefaultRouter):
     courses.register(r'aggregatedata',
                      exercise.api.csv.views.CourseAggregateDataViewSet,
                      basename='course-aggregatedata')
+    courses.register(r'mygroups',
```

Comment:
I separated all groups and "my groups" so that teachers could also see only their own groups when they submit normally. With only one API endpoint and permissions/filters, students would see their own groups like they should, but if teachers saw all groups in the course, their group selection menu for submissions would be broken.